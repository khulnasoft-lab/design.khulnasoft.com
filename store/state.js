export const getDefaultFrontmatter = () => ({
  attributes: {},
  body: '',
  lastUpdatedAt: '',
});

export default () => ({
  frontmatter: getDefaultFrontmatter(),
});
