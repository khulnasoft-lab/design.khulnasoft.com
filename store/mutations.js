import { getDefaultFrontmatter } from './state';

export default {
  resetFrontmatter(state) {
    state.frontmatter = getDefaultFrontmatter();
  },
  setFrontmatter(state, frontmatter) {
    state.frontmatter = frontmatter;
  },
};
