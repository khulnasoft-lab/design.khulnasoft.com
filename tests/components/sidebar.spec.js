import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Sidebar from '../../components/sidebar.vue';
import mutations from '../../store/mutations';
import state from '../../store/state';

const navMock = [
  {
    title: 'Top-level item 1',
    path: 'top1',
    items: [
      {
        title: 'Second-level item 1',
        path: 'second1',
      },
      {
        title: 'Second-level item 2',
        path: 'second2',
        items: [
          {
            title: 'Third-level item 1',
            path: 'third1',
          },
          {
            title: 'Third-level item 2',
            path: 'third2',
          },
        ],
      },
    ],
  },
  {
    title: 'Top-level item 2',
    path: 'top2',
    items: [
      {
        title: 'Second-level item without path',
        items: [
          {
            title: 'Third-level item 3',
            path: 'third2',
          },
        ],
      },
    ],
  },
];

jest.mock('../../nav.json', () => navMock);

const localVue = createLocalVue();
localVue.use(Vuex);

describe('sidebar component', () => {
  let wrapper;
  let store;

  const createComponent = () => {
    store = new Vuex.Store({
      state: state(),
      mutations,
    });
    wrapper = mount(Sidebar, {
      localVue,
      store,
      stubs: {
        search: true,
      },
      mocks: {
        $route: {
          fullPath: '/',
          params: {
            section: '',
          },
        },
        $router: {
          afterEach: () => {},
        },
      },
    });
  };

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders sidebar properly', () => {
    expect(wrapper.find('nav').element).toMatchSnapshot();
  });
});
