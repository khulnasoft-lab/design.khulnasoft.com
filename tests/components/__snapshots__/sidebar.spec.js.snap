// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar component renders sidebar properly 1`] = `
<nav
  aria-labelledby="nav-heading"
  class="sidebar__nav"
>
  <h2
    class="gl-sr-only"
    id="nav-heading"
  >
    Main navigation
  </h2>
   
  <ul>
    <li>
      <div
        aria-hidden="true"
        class="gl-py-2 gl-pl-4 gl-pr-3 gl-text-sm gl-font-bold"
      >
        
    Top-level item 1
  
      </div>
       
      <ul
        aria-label="Top-level item 1"
      >
        <li
          class="sidebar__nav-option"
        >
          <nuxt-link-stub
            class="sidebar__nav-anchor"
            to="/top1/second1"
          >
            
    Second-level item 1
  
          </nuxt-link-stub>
        </li>
        <li>
          <button
            aria-controls="item-second2"
            class="sidebar__nav-toggle"
          >
            <span
              class="sidebar__nav-toggle-inner"
            >
              <span>
                Second-level item 2
              </span>
               
              <svg>
                <use
                  xlink:href="#chevron-right"
                />
              </svg>
            </span>
          </button>
           
          <ul
            aria-label="Second-level item 2"
            class="sidebar__nav--depth-1"
            id="item-second2"
            style="display: none;"
          >
            <li
              class="sidebar__nav-option"
            >
              <nuxt-link-stub
                class="sidebar__nav-anchor"
                to="/top1/second2/third1"
              >
                
    Third-level item 1
  
              </nuxt-link-stub>
            </li>
            <li
              class="sidebar__nav-option"
            >
              <nuxt-link-stub
                class="sidebar__nav-anchor"
                to="/top1/second2/third2"
              >
                
    Third-level item 2
  
              </nuxt-link-stub>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <div
        aria-hidden="true"
        class="gl-py-2 gl-pl-4 gl-pr-3 gl-text-sm gl-font-bold"
      >
        
    Top-level item 2
  
      </div>
       
      <ul
        aria-label="Top-level item 2"
      >
        <li>
          <button
            aria-controls="item-undefined"
            class="sidebar__nav-toggle"
          >
            <span
              class="sidebar__nav-toggle-inner"
            >
              <span>
                Second-level item without path
              </span>
               
              <svg>
                <use
                  xlink:href="#chevron-right"
                />
              </svg>
            </span>
          </button>
           
          <ul
            aria-label="Second-level item without path"
            class="sidebar__nav--depth-1"
            id="item-undefined"
            style="display: none;"
          >
            <li
              class="sidebar__nav-option"
            >
              <nuxt-link-stub
                class="sidebar__nav-anchor"
                to="/top2/third2"
              >
                
    Third-level item 3
  
              </nuxt-link-stub>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;
